src/badguy/zeekling.cpp

```diff
@@ -176,15 +175,20 @@ Zeekling::active_update(float dt_sec) {
   if (state == FLYING) {
     if (should_we_dive()) {
       state = DIVING;
-      m_physic.set_velocity_y(2*fabsf(m_physic.get_velocity_x()));
+      dive_so_far = 0.05;
+      m_physic.set_velocity_y(dive_so_far * fabsf(m_physic.get_velocity_x()));
       m_sprite->set_action(m_dir == Direction::LEFT ? "diving-left" : "diving-right");
+      SoundManager::current()->play("sounds/deathd.wav");
     }
     BadGuy::active_update(dt_sec);
     return;
   } else if (state == DIVING) {
+    dive_so_far += 0.1;
```

Comment:
The way you did the curve makes the diving look quite terrible and not as impactful, seen as the way it hits Tux also is affected.